overlord/snapstate/snapstate.go

```diff
@@ -511,20 +511,29 @@ func defaultContentPlugProviders(st *state.State, info *snap.Info) []string {
 // validateFeatureFlags validates the given snap only uses experimental
 // features that are enabled by the user.
 func validateFeatureFlags(st *state.State, info *snap.Info) error {
-	if len(info.Layout) == 0 {
-		return nil
-	}
-
 	tr := config.NewTransaction(st)
-	var featureFlagLayouts bool
-	if err := tr.GetMaybe("core", "experimental.layouts", &featureFlagLayouts); err != nil {
-		return err
+
+	if len(info.Layout) > 0 {
+		var featureFlagLayouts bool
+		if err := tr.GetMaybe("core", "experimental.layouts", &featureFlagLayouts); err != nil {
+			return err
+		}
+		if !featureFlagLayouts {
+			return fmt.Errorf("cannot use experimental 'layouts' feature, set option 'experimental.layouts' to true and try again")
+		}
 	}
-	if featureFlagLayouts {
-		return nil
+
+	if info.InstanceKey != "" {
+		var featureFlagParallelInstall bool
```

Comment:
I'd just have `var flag bool` but this is nitpicky

Reply:
This is done now.